pkgs/tools/typesetting/tex/texlive/UPGRADING.md

```diff
@@ -36,7 +36,7 @@ See https://tug.org/texlive/acquire-mirror.html for instructions.
 
 ```bash
 curl -L http://mirror.ctan.org/tex-archive/systems/texlive/tlnet/tlpkg/texlive.tlpdb.xz \
-         | xzcat | uniq -u | sed -rn -f ./tl2nix.sed > ./pkgs.nix
+         | xzcat | uniq | sed -rn -f ./tl2nix.sed | uniq > ./pkgs.nix
```

Comment:
This looks fishy

Reply:
The second `uniq` is for packages with multiple hyphenation patterns (you get several `hasHyphens = true;` in a row). I should clarify this in the `tl2nix.sed`. At any rate, duplicates in `pkgs.nix` have no semantic effect for nix, so this is cosmetic.

I don't know about the first `uniq`. Can I just remove it? `texlive.tlpdb` does not have duplicate lines (at least, not in its current form). Certainly `-u` seems risky to me since it would erase both original and duplicate.

Reply:
I see that the content before the second uniq ends with
```
};
};
}
```

Reply:
I can't reproduce it here, even if I remove both `uniq`'s. I am running this on `texlive.tlpdb.xz` from 2021/04/08, also 2021/06/19, with `xz` and `gnused` from the unstable channel, on Darwin. There must be something weird with the final newlines that gets interpreted differently -- what system are you running this on?

Reply:
```
# nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/0747387223edf1aa5beaedf48983471315d95e16.tar.gz --pure -p coreutils curl gnused --run 'curl -L http://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2021/texlive-20210325-tlpdb-full.tar.gz | zcat | uniq | sed -rn -f ./tl2nix.sed | tail'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3020k  100 3020k    0     0  2436k      0  0:00:01  0:00:01 --:--:-- 2436k
"zztex" = {
  revision = 55862;
  stripPrefix = 0;
  sha512.run = "eb325564ee9e001e0e5ff79b6e3cc9f22f5b9b100ef021552f54953802edbce1e4b1295ff4590832f959386db7a63aaf8e31dc2d61b38e5c8a140082ca423e83";
  sha512.doc = "bd2a3a42b62da61db572058b4731161425545374584a575607f11bb2302aa2ce900b694aaf06562f4c9eb9046c80c4fbf5b50233f6d0526fac787c213070d542";
  hasRunfiles = true;
  version = "17.7";
};
};
}
# git rev-parse HEAD
2774d11edab964828bb957c7afc4598fb8d4fa9f
```
If this is not caused by your PR, then no need to fix it here.

Reply:
Ah, now I can reproduce. It's because you are `zcat`-ing the entire `tar.gz` which contains hashes and signatures as well. So nothing to fix in this PR, but I can tweak the line in `UPGRADING.md` to show the actual mirrors, both snapshots and final. Make it harder to trip on this, at least.

Reply:
Oh, I had to pick up some another URL quickly because for `curl -L http://mirror.ctan.org/tex-archive/systems/texlive/tlnet/tlpkg/texlive.tlpdb.xz` I would get
```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL /ctan/tex-archive/tex-archive/systems/texlive/tlnet/tlpkg/texlive.tlpdb.xz was not found on this server.</p>
</body></html>
```
Do you get the same?

Reply:
Yes, that url is wrong, I also get 404. I have updated the urls in the documentation.

Reply:
Oh, and also deleted the superfluous `uniq` at the beginning. That's really not necessary.